Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -2420,27 +2452,54 @@ void processSGroups(RWMol *mol) {
 
 void ProcessMolProps(RWMol *mol) {
   PRECONDITION(mol, "no molecule");
-  for (auto atom : mol->atoms()) {
-    int totV;
-    if (atom->getPropIfPresent(common_properties::molTotValence, totV) &&
-        !atom->hasProp("_ZBO_H")) {
-      if (totV == 0) {
-        continue;
+  // we have to loop the ugly way because we may need to actually replace an
+  // atom
+  for (unsigned int aidx = 0; aidx < mol->getNumAtoms(); ++aidx) {
+    auto atom = mol->getAtomWithIdx(aidx);
+    int ival = 0;
+    if (atom->getPropIfPresent(common_properties::molSubstCount, ival) &&
+        ival != 0) {
+      if (!atom->hasQuery()) {
+        atom = FileParserUtils::replaceAtomWithQueryAtom(mol, atom);
+      }
+      bool gtQuery = false;
+      if (ival == -1) {
+        ival = 0;
+      } else if (ival == -2) {
+        // as drawn
+        ival = atom->getDegree();
+      } else if (ival == 6) {
+        // 6 or more
+        gtQuery = true;
```

Comment:
This should be  `>=6` 

Also, we might want to move the `replaceAtomWithQueryAtom()`, and add an "abort" condition in case `ival < -2` or `ival == 0` (I know we don't store that, but just in case).

Reply:
good catch on the == instead of >=
I fixed that an added a test to confirm it's being handled correctly

Reply:
Actually, I may have mis-interpreted this text in the spec document:
![image](https://user-images.githubusercontent.com/540511/76644196-cf6f6b80-6556-11ea-8eb3-96605d977743.png)
It could be that any value >=6 should translate to a degree query of 6.
Any thoughts on that?

Reply:
Yeah, that's what I understand too: the degree query is capped at 6, and any SUBST value equal or greater than 6 will always be translated into the same query for degree 6.

Reply:
Updating this before merging, just to have the info captured. There's an example in the BioVia chemical representation guide (http://help.accelrysonline.com/insight/2017/content/pdf_files/bioviachemicalrepresentation2017.pdf) showing that the `s6` query is "substitution count of at least six" and that it retrieves connectivities of 7 as well